File: docs/downloads/code/Fibonacci-Number-in-Different-Programming-Languages/Java/Fibonacci.java
public class Fibonacci {
    public static int fib(int n) {
        if(n <= 1) {
            return n;
        }

        int a = 0;
        int b = 1;

        for(int i = 2; i <= n; i++) {
            int c = a + b;
            a = b;
            b = c;
        }
        return b;
    }

    public static void main(String[] args) {
        int n = Integer.parseInt(args[0]);
        System.out.println("fib(" + n + ") = " + fib(n));
    }
}

File: docs/downloads/code/Fibonacci-Number-in-Different-Programming-Languages/Scala/Fibonacci.scala
object Fibonacci {
  def fib(n: Int): Int = {
    if (n <= 1) {
      return n
    }

    var a: Int = 0
    var b: Int = 1

    for (_ <- 2 to n) {
      val c = a + b
      a = b
      b = c
    }
    b
  }

  def main(args: Array[String]): Unit = {
    val n = args(0).toInt
    println(s"fib(${n}) = ${fib(n)}")
  }
}
